frontend_egui/src/widgets/framebuffer.rs

```diff
@@ -8,13 +8,40 @@ use anyhow::{bail, Result};
 use crossbeam_channel::Receiver;
 use eframe::egui;
 use eframe::egui::Vec2;
+use serde::{Deserialize, Serialize};
 use snow_core::renderer::DisplayBuffer;
+use std::fmt::Display;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
```

Comment:
This is enum has small variants so you can derive `Copy` and get rid of the `clone()` calls